TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/activities/RunningActivity.java

```diff
@@ -384,11 +385,11 @@ private void saveUuidsNamestoHashMap() {
         mTangoServiceClientNode.callGetMapUuidsService();
     }
 
-    private void getTangoPermission(String permissionType) {
+    private void getTangoPermission(String permissionType, int requestCode) {
         Intent intent = new Intent();
         intent.setAction(REQUEST_TANGO_PERMISSION_ACTION);
         intent.putExtra(EXTRA_KEY_PERMISSIONTYPE, permissionType);
-        startActivityForResult(intent, REQUEST_CODE_TANGO_PERMISSION);
+        startActivityForResult(intent, requestCode);
```

Comment:
This is when using the Java API. In our case we use the C API so I used [this](https://developers.google.com/tango/apis/c/c-user-permissions).